### PR TITLE
Coding Standards: Add visibility to methods in `tests/phpunit/tests/` root directory.

### PR DESCRIPTION
**Pull request by costdev**
Coding Standards: Add visibility to methods in `tests/phpunit/tests/` root directory.

"Visibility should always be declared"
See: https://make.wordpress.org/core/2020/03/20/updating-the-coding-standards-for-modern-php/

This commit applies the following sniffs to files in `tests/phpunit/tests` root:
- Squiz.Scope.MethodScope
- PSR2.Methods.MethodDeclaration
- PSR2.Classes.PropertyDeclaration
- Squiz.WhiteSpace.ScopeKeywordSpacing

For most methods, these now indicate `public` visibility to avoid breaking backwards compatibility.
Any remaining methods now indicate `private` visibility where appropriate.

Additional fixups in this PR:
1. Methods touched by this PR have had unnecessary underscores removed from their names.
2. In `tests/phpunit/tests/filters.php`, the `_self_removal()` method was unused. It has now been removed.
3. In `tests/phpunit/tests/shortcode.php`, the `test_registration_bad()`, `test_registration_good()` and `sub_registration()` methods had `return` statements. Tests do not `return`. This has now been removed.
4. In `tests/phpunit/tests/functions.php`, two typos referencing non-existent methods `_test_add_mime_types_svg` and `_test_add_mime_types_woff` have been corrected to `filter_mime_types_svg` and `filter_mime_types_woff` respectively.

Trac ticket: https://core.trac.wordpress.org/ticket/54177

### CODE REVIEW

**Review by jrfnl — DISMISSED** [review]
@costdev Thanks for this PR. There are a few methods which I believe can be made `private`, but otherwise looks good.

Reviewed:
* Test methods and data provider methods need to be `public`. ✔️ 
* Test fixture methods (`set_up*()`, `tear_down*()`) to comply with the visibility of the same methods in the parent class. ✔️ 
* Verified for all methods which don't start with `test_` that the visibility is correct and left inline comments to document the verification (and some suggestions).

**Note**:
The visibility for `set_up()` and `tear_down()` _should_ be `protected` to be in line with the same in PHPUnit, however, that is currently not possible as these methods in the parent `WP_UnitTestCase_Base` class are `public`.

I'd recommend changing the visibility of these methods in the `WP_UnitTestCase_Base` class in a future PR and then fixing up all method overloads in concrete test classes to match.

This would not be a BC-break as integration tests from extenders will continue to work fine as overloading a method and widening the visibility is allowed in PHP.

File: tests/phpunit/tests/user.php [hunk 1]
```diff
@@ -772,7 +772,7 @@ function test_illegal_user_logins_multisite() {
 		$this->assertCount( 0, $response['errors']->get_error_codes() );
 	}
 
-	function _illegal_user_logins_data() {
+	public function _illegal_user_logins_data() {
```
[the remark this answers is not in the document]
Comment (costdev) [reply tests/phpunit/tests/user.php:775]: Thanks! I've renamed this data provider to `data_illegal_user_logins()` and updated references to it in the latest iteration.

File: tests/phpunit/tests/media.php [hunk 2]
```diff
@@ -108,14 +108,14 @@ function test_img_caption_shortcode_short_circuit_filter() {
 	/**
 	 * Filter used in test_img_caption_shortcode_short_circuit_filter()
 	 */
-	function _return_alt_caption() {
+	public function _return_alt_caption() {
```
[the remark this answers is not in the document]
Comment (costdev) [reply tests/phpunit/tests/media.php:111]: Thanks! I've removed the underscore and updated references in latest iteration.

File: tests/phpunit/tests/kses.php [hunk 3]
```diff
@@ -474,7 +474,7 @@ public function test_hackers_attacks() {
 		}
 	}
 
-	function _wp_kses_allowed_html_filter( $html, $context ) {
+	public function _wp_kses_allowed_html_filter( $html, $context ) {
```
[the remark this answers is not in the document]
Comment (costdev) [reply tests/phpunit/tests/kses.php:477]: Thanks! I've removed the underscore and updated references in latest iteration.

File: tests/phpunit/tests/cron.php [hunk 4]
```diff
@@ -387,7 +387,7 @@ function test_pre_schedule_event_filter() {
 	/**
 	 * Filter the scheduling of events to use the preflight array.
 	 */
-	function _filter_pre_schedule_event_filter( $null, $event ) {
+	public function _filter_pre_schedule_event_filter( $null, $event ) {
```
[the remark this answers is not in the document]
Comment (costdev) [reply tests/phpunit/tests/cron.php:390]: Thanks! I've removed the underscore and updated references in latest iteration.

File: tests/phpunit/tests/query.php [hunk 5]
```diff
@@ -78,7 +78,7 @@ function test_tag_queried_object() {
 		remove_action( 'pre_get_posts', array( $this, '_tag_queried_object' ), 11 );
 	}
 
-	function _tag_queried_object( &$query ) {
+	public function _tag_queried_object( &$query ) {
```
[the remark this answers is not in the document]
Comment (costdev) [reply tests/phpunit/tests/query.php:81]: Thanks! I've removed the underscore and updated references in the latest iteration.

File: tests/phpunit/tests/filters.php [hunk 6]
```diff
@@ -264,15 +264,15 @@ function test_filter_ref_array_result() {
 
 	}
 
-	function _self_removal( $tag ) {
+	public function _self_removal( $tag ) {
```
[the remark this answers is not in the document]
Comment (costdev) [reply tests/phpunit/tests/filters.php:267]: Thanks! I've reviewed the similar method and test in `tests/actions.php`, but I'm not entirely sure how to go about adapting this. Suggesting that this is postponed until either later in this PR or for a future PR.

File: tests/phpunit/tests/shortcode.php [hunk 7]
```diff
@@ -417,33 +417,33 @@ function data_test_strip_shortcodes() {
...
 	 */
-	function test_strip_shortcodes_filter() {
+	public function test_strip_shortcodes_filter() {
 		add_filter( 'strip_shortcodes_tagnames', array( $this, '_filter_strip_shortcodes_tagnames' ) );
 		$this->assertSame( 'beforemiddle [footag]after', strip_shortcodes( 'before[gallery]middle [footag]after' ) );
 		remove_filter( 'strip_shortcodes_tagnames', array( $this, '_filter_strip_shortcodes_tagnames' ) );
 	}
 
-	function _filter_strip_shortcodes_tagnames() {
+	public function _filter_strip_shortcodes_tagnames() {
```
[the remark this answers is not in the document]
Comment (costdev) [reply tests/phpunit/tests/shortcode.php:433]: Thanks! I've removed the underscore and updated references in latest iteration.

File: tests/phpunit/tests/shortcode.php [hunk 8]
```diff
@@ -417,33 +417,33 @@ function data_test_strip_shortcodes() {
...
+	public function _filter_atts( $out, $pairs, $atts ) {
 		$this->filter_atts_out   = $out;
 		$this->filter_atts_pairs = $pairs;
 		$this->filter_atts_atts  = $atts;
 		return $out;
 	}
 
 	// Filter shortcode atts in various ways.
-	function _filter_atts2( $out, $pairs, $atts ) {
+	public function _filter_atts2( $out, $pairs, $atts ) {
```
[the remark this answers is not in the document]
Comment (costdev) [reply tests/phpunit/tests/shortcode.php:446]: Thanks! I've removed the underscore and updated references in latest iteration.

File: tests/phpunit/tests/link.php [hunk 9]
```diff
@@ -4,14 +4,14 @@
  */
 class Tests_Link extends WP_UnitTestCase {
 
-	function _get_pagenum_link_cb( $url ) {
+	public function _get_pagenum_link_cb( $url ) {
```
[the remark this answers is not in the document]
Comment (jrfnl) [reply tests/phpunit/tests/link.php:7]: Hmm... looks like that commit is missing....

File: tests/phpunit/tests/filters.php (hunk 6 above)
[the remark this answers is not in the document]
Comment (jrfnl) [reply tests/phpunit/tests/filters.php:267]: Let's open an issue about it and remove the unused function for now.

File: tests/phpunit/tests/rewrite.php [hunk 10]
```diff
@@ -160,7 +160,7 @@ function test_url_to_postid_of_http_site_when_current_site_uses_https() {
 	 * @param int|null    $blog_id     Site ID, or null for the current site.
 	 * @return string                  The complete home URL including scheme and path.
 	 */
-	function _filter_http_home_url( $url, $path, $orig_scheme, $_blog_id ) {
+	public function _filter_http_home_url( $url, $path, $orig_scheme, $_blog_id ) {
```
[the remark this answers is not in the document]
Comment (jrfnl) [reply tests/phpunit/tests/rewrite.php:163]: Hmm... looks like that commit is missing....

File: tests/phpunit/tests/media.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (jrfnl) [reply tests/phpunit/tests/media.php:111]: Hmm... looks like that commit is missing....

File: tests/phpunit/tests/link.php (hunk 9 above)
[the remark this answers is not in the document]
Comment (costdev) [reply tests/phpunit/tests/link.php:7]: Nope, incoming, just reviewing the changes :)

File: tests/phpunit/tests/shortcode.php (hunk 7 above)
[the remark this answers is not in the document]
Comment (jrfnl) [reply tests/phpunit/tests/shortcode.php:433]: Hmm... looks like that commit is missing....

File: tests/phpunit/tests/shortcode.php [hunk 11]
```diff
@@ -417,33 +417,33 @@ function data_test_strip_shortcodes() {
...
 	}
 
-	function _filter_strip_shortcodes_tagnames() {
+	public function _filter_strip_shortcodes_tagnames() {
 		return array( 'gallery' );
 	}
 
 	// Store passed in shortcode_atts_{$shortcode} args.
-	function _filter_atts( $out, $pairs, $atts ) {
+	public function _filter_atts( $out, $pairs, $atts ) {
```
[the remark this answers is not in the document]
Comment (jrfnl) [reply tests/phpunit/tests/shortcode.php:438]: Hmm... looks like that commit is missing....

File: tests/phpunit/tests/shortcode.php (hunk 8 above)
[the remark this answers is not in the document]
Comment (jrfnl) [reply tests/phpunit/tests/shortcode.php:446]: Hmm... looks like that commit is missing....

File: tests/phpunit/tests/user.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (jrfnl) [reply tests/phpunit/tests/user.php:775]: Hmm... looks like that commit is missing....

[the remark this answers is not in the document]
Comment (hellofromtonya) [reply tests/phpunit/tests/user.php]: I confirmed that this change was indeed done https://github.com/WordPress/wordpress-develop/blob/dfa581a91186c901e8a9969b75439aa5b67fa514/tests/phpunit/tests/user.php#L775. Marking this as resolved.

**Review by hellofromtonya — APPROVED** [review]
Looks good. All changes @jrfnl requested are done and verified. Ready for `commit`.

**Comment by hellofromtonya** [comment]
Committed via https://core.trac.wordpress.org/changeset/52010.

**Pull request closed without merging**